pyg_lib/csrc/sampler/cpu/dist_neighbor_kernel.cpp

```diff
@@ -0,0 +1,261 @@
+#include <ATen/ATen.h>
+#include <ATen/Parallel.h>
+#include <torch/library.h>
+
+#include "parallel_hashmap/phmap.h"
+
+#include "pyg_lib/csrc/sampler/cpu/mapper.h"
+#include "pyg_lib/csrc/utils/cpu/convert.h"
+#include "pyg_lib/csrc/utils/types.h"
+
+namespace pyg {
+namespace sampler {
+
+namespace {
+
+template <typename scalar_t>
+std::tuple<at::Tensor, at::Tensor> get_sampled_edges(
+    std::vector<scalar_t> sampled_rows,
+    std::vector<scalar_t> sampled_cols,
+    const bool csc = false) {
+  const auto row = pyg::utils::from_vector(sampled_rows);
+  const auto col = pyg::utils::from_vector(sampled_cols);
+
+  if (!csc) {
+    return std::make_tuple(row, col);
+  } else {
+    return std::make_tuple(col, row);
+  }
+}
+
+template <bool disjoint>
+std::tuple<at::Tensor, at::Tensor> relabel(
+    const at::Tensor& seed,
+    const at::Tensor& sampled_nodes_with_dupl,
+    const std::vector<int64_t>& sampled_nbrs_per_node,
+    const int64_t num_nodes,
+    const c10::optional<at::Tensor>& batch,
+    const bool csc) {
+  if (disjoint) {
+    TORCH_CHECK(batch.has_value(),
+                "Batch needs to be specified to create disjoint subgraphs");
+    TORCH_CHECK(batch.value().is_contiguous(), "Non-contiguous 'batch'");
+    TORCH_CHECK(batch.value().numel() == sampled_nodes_with_dupl.numel(),
+                "Each node must belong to a subgraph.'");
+  }
+  TORCH_CHECK(seed.is_contiguous(), "Non-contiguous 'seed'");
+  TORCH_CHECK(sampled_nodes_with_dupl.is_contiguous(),
+              "Non-contiguous 'sampled_nodes_with_dupl'");
+
+  at::Tensor out_row, out_col;
```

Comment:
I believe creating an empty vector is cheaper than creating a Tensor. You could declare here `sampled_rows` and `sampled_cols` and at the end simply return `get_sampled_edges<scalar_t>(sampled_rows, sampled_cols, csc);`

Reply:
scalar_t will not work outside of the macro